### DISCUSSION TITLE
How can I reuse a mock of a module from node_module ?

### DISCUSSION THREAD

**Discussion by robbporto** [vitest-dev/vitest#5316] [Q&A] [answered]
I have the following code (it's working fine):

```js
import {render} from '@testing-library/react'
import {vi} from 'vitest'
import LoginPage from 'LoginPage.jsx'

const mockUseNavigate = vi.fn()

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom')
  return {
    ...actual,
    useNavigate: () => mockUseNavigate
  }
})

it('should redirect to the home page', () => {
  render(<LoginPage/>)

  expect(mockUseNavigate).toHaveBeenCalledWith('/home')
})
```

However, I have other tests that I need `useNavigate` to be mocked. I don't want to repeat `vi.mock` across a bunch of tests, ideally I would have the `vi.mock` part in a file and just call that file inside any test I want to mock `useNavigate`.

How can I do that?

Thank you!

**Comment by sheremet-va** [comment, marked as answer]
You can put the mock statement in a file and make sure to always import it first before other imports

**Reply by robbporto** [reply]
Wow. That was fast. Thank you!

Just another question: is there any sort of convention of where I should put this mock file? In Jest, we normally put inside a `__mocks__` folder at the top level when mocking node modules.

**Reply by sheremet-va** [reply]
That's also how it works in Vitest

**Reply by robbporto** [reply]
Thank you, @sheremet-va. Really awesome support.
